src/js/webSerial.js

```diff
@@ -75,7 +74,7 @@ class WebSerial extends EventTarget {
 
     createPort(port) {
         return {
-            path: `D${this.port_counter++}`,
+            path: `serial_${this.port_counter++}`,
```

Comment:
Are we able to just use "serial" as path here. We had `connectionID` before for as a means to check we had established a new serial connection.

port_counter => portCounter.

Reply:
This is only an internal id, I modified it to make clear it comes from serial. 
We don't have any connection id in the api.

Reply:
We no longer use `connectionType`.

Now `connectionId` is not assigned in WebSerial:

```
SERIAL:  Connection opened with ID: undefined, Baud: 115200
```

```
src/js/serial_backend.js:273:        gui_log(i18n.getMessage('serialPortOpened', serial.connectionType === 'serial' ? [serial.connectionId] : [openInfo.socketId]));
src/js/serial_backend.js:797:    const isVirtual = CONFIGURATOR.virtualMode && GUI.connected_to == 'virtual' && CONFIGURATOR.connectionValid && serial.connectionId === 'virtual';
src/js/serial_backend.js:806:            console.log(`Disconnecting from: ${serial.connectionId}`, GUI.connected_to);
src/js/main.js:116:    const connectionId = serial.connectionId;
src/js/main.js:118:    if (connectionId && CONFIGURATOR.connectionValid && !CONFIGURATOR.virtualMode) {
src/js/main.js:133:        sendFn(connectionId, bufferOut, function () {
src/js/main.js:160:            sendFn(connectionId, bufferOut, function () {
src/js/main.js:164:    } else if (connectionId) {
src/js/tabs/firmware_flasher.js:1314:    if (serial.connected || serial.connectionId) {
src/js/tabs/firmware_flasher.js:1315:        console.warn('Attempting to connect while there still is a connection', serial.connected, serial.connectionId, serial.openCanceled);
src/js/serial.js:14:    connectionId:   false,
src/js/serial.js:49:                self.connectionId = connectionInfo.connectionId;
src/js/serial.js:63:                                chrome.serial.setPaused(self.connectionId, false, function () {
src/js/serial.js:85:                                chrome.serial.setPaused(info.connectionId, false, function() {
src/js/serial.js:123:                console.log(`${self.connectionType}: connection opened with ID: ${connectionInfo.connectionId} , Baud: ${connectionInfo.bitrate}`);
src/js/serial.js:134:                    self.connectionId = connectionInfo.connectionId;
src/js/serial.js:135:                    console.log(`${self.connectionType}: connection opened with ID: ${connectionInfo.connectionId} , but request was canceled, disconnecting`);
src/js/serial.js:170:                self.connectionId = createInfo.socketId;
src/js/serial.js:184:                                    if (info.socketId !== self.connectionId) return;
src/js/serial.js:213:            self.connectionId = 'virtual';
src/js/serial.js:224:        const id = self.connectionId;
src/js/serial.js:228:        if (self.connectionId) {
src/js/serial.js:241:                    chrome.sockets.tcp.disconnect(self.connectionId, function () {
src/js/serial.js:248:                disconnectFn(self.connectionId, function (result) {
src/js/serial.js:260:            self.connectionId = false;
src/js/serial.js:296:        chromeType.getInfo(this.connectionId, callback);
src/js/serial.js:308:                console.log(`${self.connectionType}: attempting to send when disconnected. ID: ${self.connectionId}`);
src/js/serial.js:320:            sendFn(self.connectionId, _data, function (sendInfo) {
src/js/serial.js:436:                    self.connectionId = false;
src/js/serial.js:459:        console.warn(`${self.connectionType}: ${resultMessage} ID: ${self.connectionId} (${direction})`);
src/js/protocols/stm32.js:872:            if (serial.connectionId) {
src/js/webSerial.js:126:            this.connectionId = connectionInfo.connectionId;
src/js/webSerial.js:136:                `${this.logHead} Connection opened with ID: ${connectionInfo.connectionId}, Baud: ${options.baudRate}`,
src/js/webSerial.js:154:            this.connectionId = connectionInfo.connectionId;
src/js/webSerial.js:157:                `${this.logHead} Connection opened with ID: ${connectionInfo.connectionId}, but request was canceled, disconnecting`,
src/js/webSerial.js:207:                `${this.logHead}Connection with ID: ${this.connectionId} closed, Sent: ${this.bytesSent} bytes, Received: ${this.bytesReceived} bytes`,
src/js/webSerial.js:210:            this.connectionId = false;
src/js/webSerial.js:216:                `${this.logHead}Failed to close connection with ID: ${this.connectionId} closed, Sent: ${this.bytesSent} bytes, Received: ${this.bytesReceived} bytes`,
src/js/msp.js:312:        const connected = isWeb() ? serial.connected : serial.connectionId;
src/js/msp.js:342:                console.warn(`MSP: data request timed-out: ${code} ID: ${serial.connectionId} TAB: ${GUI.active_tab} TIMEOUT: ${this.timeout} QUEUE: ${this.callbacks.length} (${this.callbacks.map((e) => e.code)})`);
```